discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -73,5 +76,8 @@ def register_collector(cls, name, collector_func):
 
     @classmethod
     def populate_collectors(cls):
-        for collector in cls.registered_collectors.values():
-            collector()
+        for name, collector in enumerate(cls.registered_collectors):
```

Comment:
I simply use enumerate here so that I can loop through both the `name` and `collector()`. If there's a failure, I'll be able to use `name` in my logs which will include a full stack trace that will be useful in debugging the collector.

Reply:
understood - per msg, it seems like enumerate will not return `name, collector` but rather `index, name` ?

Reply:
Indeed! I've updated the code to be `.items()` and not `enumerate()`.